src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -3117,6 +3286,296 @@ NetworkOPsImp::unsubAccountInternal(
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+            };
+
+            /*
+             * search backward until the genesis ledger or asked to stop
+             */
+            while (lastLedgerSeq >= 2 && !subInfo.index_->stopHistorical_)
+            {
+                auto startLedgerSeq =
+                    (lastLedgerSeq > 256 + 2 ? lastLedgerSeq - 256 : 2);
+                JLOG(m_journal.trace())
+                    << "AccountHistory job for account " << toBase58(accountId)
+                    << ", working on ledger range [" << startLedgerSeq << ","
+                    << lastLedgerSeq << "]";
+
+                auto haveRange = [&]() -> bool {
+                    std::uint32_t validatedMin = UINT_MAX;
+                    std::uint32_t validatedMax = 0;
+                    auto haveSomeValidatedLedgers =
+                        app_.getLedgerMaster().getValidatedRange(
+                            validatedMin, validatedMax);
+
+                    return haveSomeValidatedLedgers &&
+                        validatedMin <= startLedgerSeq &&
+                        lastLedgerSeq <= validatedMax;
+                }();
+
+                if (!haveRange)
+                {
+                    JLOG(m_journal.debug())
+                        << "AccountHistory reschedule job for account "
+                        << toBase58(accountId) << ", incomplete ledger range ["
+                        << startLedgerSeq << "," << lastLedgerSeq << "]";
+                    setAccountHistoryJobTimer(subInfo);
+                    return;
+                }
+
+                RelationalDBInterface::AccountTxPageOptions options = {
+                    accountId, startLedgerSeq, lastLedgerSeq, {}, 0, true};
+
+                auto [txns, marker] =
+                    dynamic_cast<RelationalDBInterfaceSqlite*>(
+                        &app_.getRelationalDBInterface())
+                        ->newestAccountTxPage(options);
+
+                for (auto const& [tx, meta] : txns)
+                {
+                    if (!tx || !meta)
+                    {
+                        send(rpcError(rpcINTERNAL), true);
+                        return;
+                    }
+                    auto curTxLedger =
+                        app_.getLedgerMaster().getLedgerBySeq(tx->getLedger());
+                    if (!curTxLedger)
+                    {
+                        send(rpcError(rpcINTERNAL), true);
+                        return;
+                    }
+                    std::shared_ptr<STTx const> stTxn = tx->getSTransaction();
+                    if (!stTxn)
+                    {
+                        send(rpcError(rpcINTERNAL), true);
+                        return;
+                    }
+                    Json::Value jvTx = transJson(
+                        *stTxn, meta->getResultTER(), true, curTxLedger);
```

Comment:
I think you can just call `stTxn->getJson()`, and then you don't need to pass in the ledger. Then you can avoid looking up the ledger from `LedgerMaster` for every transaction

Reply:
In `transJson()` it calls `getJson()` and adds more fields. Some of the fields are needed.